onnxruntime/test/platform/threadpool_test.cc

```diff
@@ -38,7 +40,7 @@ std::unique_ptr<TestData> CreateTestData(int num) {
 }
 
 void IncrementElement(TestData& test_data, ptrdiff_t i) {
-  std::lock_guard<onnxruntime::OrtMutex> lock(test_data.mutex);
+  absl::MutexLock lock(&test_data.mutex);
```

Comment:
Is it possible to use `std::lock_guard`?

Reply:
No. Because lock_guard calls "lock()"/"unlock()" instead of "Lock()"/"Unlock()".

Reply:
See, if we inhert OrtMutex from absl::Mutex or wrap it, it would be possible to avoid most of the changes and keep standard lock_guard.

Reply:
But that's an extra layer, which may incur unnecessary cost. Not only we will use absl's mutex and condvar, we may also leverage their other synchronization primitives like Barrier. From that point of view, std::lock_guard doesn't seem to be better than absl::MutexLock.

Reply:
There is not an extra layer. It will all be taken care of with inlining and what not. The same thing can be done with Condition var and everything else, then the changes will be largely confined to one file.

Reply:
ABSL's condvar's interface is very different than the standard one. It doesn't have a notify() function.

Reply:
Absl::Mutex's Unblock() function serves as the notify() function for condvars.

Reply:
Fine. I will abandon this change.